test/recipes/30-test_evp_data/evppkey_ecdsa_sigalg.txt

```diff
@@ -0,0 +1,238 @@
+AMSdNgiG5wSTamZ44ROdJreBn36QBEEE5JcIvn36opqjEm/k59Al40rBAxWM2TPG
+l0L13Je51zHpfXQ9Z2o7IQicMXP4wSfJ0qCgg2bgydqoxlYrlLGuVQIhAP////8A
+AAAA//////////+85vqtpxeehPO5ysL8YyVRAgEBBG0wawIBAQQgec92jwduadCk
+OjoNRI+YT5Be5TkzZXzYCyTLkMOikDmhRANCAATtECEhQbLEaiUj/Wu0qjcr81lL
+46dx5zYgArz/iaSNJ3W80oO+F7v04jlQ7wxQzg96R0bwKiMeq5CcW9ZFt6xg
+-----END PRIVATE KEY-----
+
+PrivateKey = B-163
+-----BEGIN PRIVATE KEY-----
+MGMCAQAwEAYHKoZIzj0CAQYFK4EEAA8ETDBKAgEBBBUDnQW0mLiHVha/jqFznX/K
+DnVlDgChLgMsAAQB1qZ00fPIct+QN8skv1XIHtBNp3EGLytJV0tsAUTYtGhtrzRj
+e3GzYyg=
+-----END PRIVATE KEY-----
+
+PrivateKey = secp256k1
+-----BEGIN PRIVATE KEY-----
+MIGEAgEAMBAGByqGSM49AgEGBSuBBAAKBG0wawIBAQQgsLpFV9joHc0bisyV53XL
+mrG6/Gu6ZaHoXtKP/VFX44ehRANCAARLYWGgp5nP4N8guypLSbYGCVN6ZPCnWW4x
+srYkcpdbxr4neRT3zC62keCKgPbJf5SIHkJ2Tcaw6hVSrBOUFtix
+-----END PRIVATE KEY-----
+
+Title = FIPS tests
+
+# Test that a nist curve with < 112 bits is allowed in fips mode for verifying
+Verify-Message = ECDSA-SHA256:B-163
+Input = "Hello World"
+Output = 302e0215027bb891747468b4b59ca2a2bf8f42d29d08866cf5021502cc311b25e9a2168e42240b07a6071070f687eb3b
+
+# Test that a nist curve with SHA3 is allowed in fips mode
+# The sign will get a mismatch error since the output signature changes on each run 
+Sign-Message = ECDSA-SHA3-512:P-256
+Input = "Hello World"
+Result = KEYOP_MISMATCH
+
+# Test that a explicit curve that is a named curve is allowed in fips mode
+Verify-Message = ECDSA-SHA256:P-256_NAMED_CURVE_EXPLICIT
+Input = "Hello World"
+Output = 30450220796fcf472882ed5779226dcd0217b9d2b9acfe4fa2fb0109c8ee63c63adc1033022100e306c69f7e31b9a5d54eb12ba813cddf4de4af933e4f6cea38a0817d9d831d91
+
+Title = FIPS Negative tests (using different curves and digests)
+
+# Test that a explicit curve is not allowed in fips mode
+Availablein = fips
+Verify-Message = ECDSA-SHA256:EC_EXPLICIT
+Input = "Hello World"
+Result = KEYOP_INIT_ERROR
+
+# Test that a curve with < 112 bits is not allowed in fips mode for signing
+Availablein = fips
+Sign-Message = ECDSA-SHA3-512:B-163
+Securitycheck = 1
+Input = "Hello World"
+Result = KEYOP_INIT_ERROR
+
+# Test that a non nist curve is not allowed in fips mode
+Availablein = fips
+Sign-Message = ECDSA-SHA3-512:secp256k1
+Securitycheck = 1
+Input = "Hello World"
+Result = KEYOP_INIT_ERROR
+
+# Test that SHA1 is not allowed in fips mode for signing
+Availablein = fips
+Sign-Message = ECDSA-SHA1:B-163
+Securitycheck = 1
+Input = "Hello World"
+Result = KEYOP_INIT_ERROR
+
+# Test that SHA1 is not allowed in fips mode for signing
+Availablein = fips
+Sign = ECDSA-SHA1:P-256
+Securitycheck = 1
+Input = "0123456789ABCDEF1234"
+Result = KEYOP_INIT_ERROR
+
+# Invalid non-approved digest
+Availablein = fips
+Verify-Message = ECDSA-MD5:P-256-PUBLIC
+Securitycheck = 1
+Result = KEYOP_INIT_ERROR
+
+Title = FIPS Indicator tests
+# Check that the indicator callback is triggered
+# We check for signature mismatch since the signature is unique
+
+FIPSversion = >=3.4.0
```

Comment:
May as well add the Availablein = fips
To most places with FIPSVersion = >=3.4.0 in both files...

Reply:
That would be incorrect.  It *is* available in the default provider too, and should be tested there as well!

Reply:
Those tests are for FIPS indicators which are only present in the FIPS module, so it will be correct once the FIPSVersion skipping problem is merged.

Reply:
Okie